bundles/io/org.eclipse.smarthome.io.rest.sitemap.test/src/test/java/org/eclipse/smarthome/io/rest/sitemap/internal/SitemapResourceTest.java

```diff
@@ -270,4 +382,384 @@ public TestItem(String name) {
             return Collections.emptyList();
         }
     }
+
+    public class TestEClass implements EClass {
```

Comment:
Can't you use a mock of `EClass` instead of this lengthy class definition here?

Reply:
Somethink like that ?
```
EClass testEClass = mock(EClass.class);
when(testEClass .getName()).thenReturn("slider");
Widget w1 = mock(Widget.class);
when(w1.eClass()).thenReturn(testEClass);
```
Is it possible to mock an interface class ?